### DISCUSSION TITLE
Why Feed forward put X_test first?

### DISCUSSION THREAD

**Discussion by DuyDoanLearning** [mrdbourke/pytorch-deep-learning#523] [Q&A]
Hi guys, thanks guys for reading my question, and also, thanks Daniel, for making such a fantastic video.

I have one question. I am in the part of making Linear Regression. Typically, with sklearn, we fit the training dataset to the model and make the model learn from the training data first. Then with the rules known, it will predict with the test set.

However, for the model in PyTorch, while building a training loop, we feed forward the X_test first. And then we optimize to fit the test data. I see that when we define the Linear Model, we just put in random parameters, so I don't see any connection with the training data here? So what is the use of training data in PyTorch model?

Thanks for helping me, hope you guys have a good day ahead.
